lib/tree_searcher.rb

```diff
@@ -0,0 +1,57 @@
+require_relative 'dom_parser.rb'
+require_relative 'node_renderer.rb'
+
+# TODO: implement search_descendents and search_parents
+
+class TreeSearcher
+
+  attr_reader :matches
+
+  def initialize
+    @dom_parser = DomParser.new
+    @tree = @dom_parser.build_tree()
+    @matches = []
+  end
+
+  def search_by(attribute, search_term)
+    puts "You searched for: #{attribute}, #{search_term}"
+    find_matches(attribute, search_term)
+    puts "There are #{@matches.length} tags that match your search."
+    render_matches
+  end
+
+
+  def render_matches
```

Comment:
You use a very similar method in your render class which you good just give a parameter and use for rendering this as well